packages/angular/projects/ui-angular/src/lib/components/authenticator/components/setup-totp/setup-totp.component.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { AuthenticatorService } from '../../../../services/authenticator.service';
+import { SetupTotpComponent } from './setup-totp.component';
+import { AmplifySlotComponent } from '../../../../utilities/amplify-slot/amplify-slot.component';
+import { ErrorComponent } from '../../../../primitives/error/error.component';
+import { BaseFormFieldsComponent } from '../base-form-fields/base-form-fields.component';
+import { FormFieldComponent } from '../form-field/form-field.component';
+import { ButtonComponent } from '../../../../primitives/button/button.component';
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+import QRCode from 'qrcode';
+import { Auth } from 'aws-amplify';
+import { MockComponent } from 'ng-mocks';
+import { getTotpCode } from '@aws-amplify/ui';
+
+const mockUser = { username: 'username' };
+const mockContext = {
+  country_code: '100',
+  formFields: { setupTOTP: { QR: null } },
+  user: mockUser,
+};
+
+const DEFAULT_TOTP_ISSUER = 'AWSCognito';
+const SECRET_KEY = 'secretKey';
+
+const setupTOTPSpy = jest.spyOn(Auth, 'setupTOTP');
+
+const toDataURLSpy = jest.spyOn(QRCode, 'toDataURL');
+
+describe('SetupTotpComponent', () => {
+  let fixture: ComponentFixture<SetupTotpComponent>;
+  let component: SetupTotpComponent;
+
+  beforeEach(async () => {
+    jest.resetAllMocks();
+
+    setupTOTPSpy.mockResolvedValue(SECRET_KEY);
+
+    const mockAuthenticatorService = {
+      _state: {},
+      isPending: false,
+      authState: {
+        context: {
+          actorRef: { getSnapshot: () => ({ context: { ...mockContext } }) },
+        },
+      },
+      updateForm: jest.fn(),
+      submitForm: jest.fn(),
+      context: jest.fn().mockReturnValue({}),
+      slotContext: jest.fn().mockReturnValue({}),
+    };
+
+    await TestBed.configureTestingModule({
+      declarations: [
+        SetupTotpComponent,
+        ErrorComponent,
+        AmplifySlotComponent,
+        BaseFormFieldsComponent,
+        ButtonComponent,
+        MockComponent(FormFieldComponent),
+      ],
+      providers: [
+        { provide: AuthenticatorService, useValue: mockAuthenticatorService },
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(SetupTotpComponent);
+    component = fixture.componentInstance;
+  });
+  it('it works', () => {
+    expect(fixture).toBeTruthy();
+  });
```

Comment:
Can we give this a more meaningful name? Also the `'it'` part of `'it works'` is semantically redundant :)

Reply:
Ok. I'll change it to `it('successfully mounts')` . Which is more clear what the test does

Reply:
b226db3